driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java

```diff
@@ -161,17 +162,27 @@ private static void validateWriteRequestHint(final ConnectionDescription connect
         if (serverIsLessThanVersionThreeDotFour(connectionDescription)) {
             throw new IllegalArgumentException(format("Hint not supported by wire version: %s",
                     connectionDescription.getMaxWireVersion()));
-        } else if ((request instanceof DeleteRequest || request instanceof UpdateRequest) && !writeConcern.isAcknowledged()) {
-            throw new MongoClientException("Specifying hints with an unacknowledged WriteConcern is not supported");
+        }
+        if (!writeConcern.isAcknowledged()) {
+            if (request instanceof UpdateRequest && serverIsLessThanVersionFourDotTwo(connectionDescription)) {
+                throw new IllegalArgumentException(format("Hint not supported by wire version: %s",
```

Comment:
I can't see any rationale why this was a MongoClientException and not an IllegalArgumentException, so I changed it to be consistent.  We never documented one or the other so I don't think it's a breaking change.

Reply:
It's difficult to tell what exception should be used based on the usage of `MongoClientException`. We have `throw new MongoClientException("Read preference in a transaction must be primary")` and `throw new MongoClientException("Unsupported server API version: " + value)`. So `MongoClientException` also seems fine here given its usage in other places.